mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -100,11 +94,6 @@
 
   @Override
   public List<InputSplit> getSplits(JobContext context) {
-    if (splits != null) {
```

Comment:
Hive keeps a cache of input format instance arounds (see [`HiveInputFormat`](https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java#L112)) that breaks this logic so I've chosen to remove it for now.

We can re-implement this later but the logic will have to be a bit more robust.